src/mds/CInode.cc

```diff
@@ -4538,6 +4545,12 @@ void CInode::validate_disk_state(CInode::validated_data *results,
         }
         results->backtrace.error_str << "failed to decode on-disk backtrace ("
                                      << bl.length() << " bytes)!";
+        // we probably have a new unwritten file!
+        // so skip the backtrace scrub for this entry and say that all's well
+        if (bl.length() == 0) {
+          results->backtrace.passed = true;
+          dout(20) << "we've probably hit a new unwritten entry" << dendl;
+        }
```

Comment:
In both of these cases we can be more specific and simply check for ENOENT (I think that's the right error code?) rather than accepting *any* error and presuming it's okay.